spring-cloud-starter-stream-app-function/src/main/java/org/springframework/cloud/stream/app/function/app/FunctionProperties.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.app.function.app;
+
+import org.hibernate.validator.constraints.NotBlank;
+
+import javax.validation.constraints.Size;
+
+/**
+ * Configuration properties for deciding how to locate the functional class to execute.
+ *
+ * @author Eric Bottard
+ */
+public class FunctionProperties {
+
+	/**
+	 * Location(s) of jar archives containing the supplier/function/consumer class to run.
+	 */
+	@Size(min = 1)
+	private String[] location;
+
```

Comment:
we could go ahead and add a dependencies array here as well, as supported since this commit: https://github.com/spring-cloud/spring-cloud-function/commit/2fdfda941676d438c6b1e012b3dd35acdd5069a6

Reply:
actually that's for the compiler functionality...  in this case as @trisberg just pointed out to me, users could provide the dependencies in this same property

Reply:
ClassUtils use camel case - like here: https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/util/ClassUtils.html#resolveClassName-java.lang.String-java.lang.ClassLoader-